### PR TITLE
Omit participants from sharing table if they're only included by type and don't have SHARER.

### PR DESCRIPTION
**Pull request by lionell-pack-ttd**
Filter out participants from the sharing table if they are only included by type and don't have SHARER.
Add tests for the edge cases around SHARER.

### CODE REVIEW

File: src/web/components/SharingPermission/SharingPermissionsTable.tsx [hunk 1]
```diff
@@ -230,6 +230,7 @@ export function SharingPermissionsTable({
   const { sites, isLoading } = useAllSitesList();
   const getSharingParticipants: () => SharingSiteWithSource[] = () => {
     return sites!
+      .filter((p) => p.canBeSharedWith || sharedSiteIds.includes(p.id))
```
[the remark this answers is not in the document]
Comment (lionell-pack-ttd) [reply src/web/components/SharingPermission/SharingPermissionsTable.tsx:233]: Unfortunately in this case, we do need all sites - because they might be explicitly sharing with a participant that's not in the available list.
The rest of sharing uses the `useAvailableSiteList` call, but that could omit sites we need here.

The whole design of this section is a bit too complicated but I want to get this bug fixed without re-working it.
